cfme/tests/automate/test_instance.py

```diff
@@ -354,3 +358,130 @@ def test_quota_source_value(request, entity, search, copy_quota_instance, generi
     provision_request.wait_for_request(method='ui')
     request.addfinalizer(lambda: provision_request.remove_request(method="rest"))
     assert result.validate(wait="60s")
+
+
+@pytest.fixture(scope='module')
+def setup(domain, klass, namespace):
+    """This fixture creates common domain, namespace, two classes, instance and method. This setup
+    is common for both parameterized tests"""
+
+    klass.schema.add_fields(*[{'name': state, 'type': 'State'}
+                              for state in [STATE1, STATE2, STATE3]])
+
+    # Adding method to execute via parent instance(state machine instance)
+    method = klass.methods.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        location='inline',
+        script="""
+               \n$evm.log(:info, "Hello from method of parent instance")
+                \nexit MIQ_STOP
+                """
+    )
+
+    # Adding parent instance to call automate method and child instances in other class
+    instance = klass.instances.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        description=fauxfactory.gen_alphanumeric(),
+        fields={STATE3: {"value": f"METHOD::{method.name}"}}
+    )
+
+    # Creating other class for two child instances
+    klass2 = namespace.classes.create(
+        name=fauxfactory.gen_alpha(),
+        display_name=fauxfactory.gen_alpha(),
+        description=fauxfactory.gen_alpha()
+    )
+    klass2.schema.add_fields(*[{'name': state, 'type': 'State'}
+                               for state in [STATE1, STATE2, STATE3]])
+
+    yield klass2, instance
+    method.delete_if_exists()
+    instance.delete_if_exists()
+    klass2.delete_if_exists()
+
+
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1441353])
+@pytest.mark.parametrize("process", ["miq_stop", "miq_abort"])
+def test_miq_stop_abort_with_state_machines(request, setup, process, domain, klass, namespace):
+    """
```

Comment:
**Suggestion:** Please add some description on what your test is trying to do.

Reply:
Yes @kedark3 
I agree with you here. Actually this is very complicated test case to understand. So I am adding ```testSteps``` here

Reply:
Thanks Ganesh.